sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -897,6 +897,14 @@ ssize_t coap_opt_add_block(coap_pkt_t *pkt, coap_block_slicer_t *slicer,
     return coap_opt_add_uint(pkt, option, _slicer2blkopt(slicer, more));
 }
 
+ssize_t coap_opt_add_proxy_uri(coap_pkt_t *pkt, const char *uri)
+{
+    assert(pkt);
+    assert(uri);
+
+    return _add_opt_pkt(pkt, COAP_OPT_PROXY_URI, (uint8_t *)uri, strlen(uri));
```

Comment:
Originally, I expected to see an inline function that used `coap_opt_add_string()`. However, since the URI is just a single string, I'm OK with this choice. If the user wants to avoid it, they can use `coap_opt_add_string()` directly.

Reply:
Since `coap_opt_add_string()` brings a lot of overhead for splitting resource strings into multiple options, I chose the more general `_add_opt_pkt()`. I could've also used `coap_opt_add_opaque()`, but in the end this also just calls `_add_opt_pkt()`.